### PR TITLE
Add GyroE (E-CLP)

### PR DESCRIPTION
**Pull request by joshguha**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -115,7 +131,6 @@ export interface SwapV2 {
     assetOutIndex: number;
     amount: string;
     userData: string;
-    returnAmount?: string;
```
Comment (johngrantuk) [inline src/types.ts:118]: This needs to remain.

File: src/pools/gyroEPool/gyroEMath/gyroSignedFixedPoint.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    SQRT_1E_NEG_15,
+    SQRT_1E_NEG_17,
+    ONE_XP,
+} from './constants';
+
+/////////
+/// ARITHMETIC HELPERS
+/////////
+
+export function mulUp(a: BigNumber, b: BigNumber) {
```
Comment (johngrantuk) [inline src/pools/gyroEPool/gyroEMath/gyroSignedFixedPoint.ts:20]: I think a lot of this is repeated functions that already exist in other helpers - could you consolidate them please?

File: src/pools/index.ts [hunk 3]
```diff
@@ -60,13 +63,11 @@ export function parseNewPool(
             newPool.setCurrentBlockTimestamp(currentBlockTimestamp);
         } else if (pool.poolType.toString().includes('Linear'))
             newPool = LinearPool.fromPool(pool);
-        else if (
-            pool.poolType === 'StablePhantom' ||
-            pool.poolType === 'ComposableStable'
-        )
+        else if (pool.poolType === 'StablePhantom')
```
Comment (johngrantuk) [inline src/pools/index.ts:66]: We need this to remain. (Maybe you need to rebase from develop?)

File: src/types.ts [hunk 4]
```diff
@@ -61,7 +62,6 @@ export interface Swap {
     maxPrice?: string;
     tokenInDecimals: number;
     tokenOutDecimals: number;
-    returnAmount?: string;
```
Comment (johngrantuk) [inline src/types.ts:64]: This needs to remain

**Comment by johngrantuk** [comment]
@joshguha - see some comments on some issues that need to be addressed inline.

Also could you add a test against a local fork (you could use your Goerli test pool) for the new pool type similar to how we did it here: `test/gyro2.integration.spec.ts`. This will allow us to confirm everything matches the onchain result. Thanks!

**Comment by joshguha** [comment]
Thanks @johngrantuk, I've...
- returned the removed code so that the `yarn build` command works
- consolidated the Gyro helper functions into a separate file
- added an integration test for the GyroE pool using the Goerli deployment

**Review by fsxforte — COMMENTED** [review]
Fixed URLs

**Review by johngrantuk — APPROVED** [review]

**Review by sergioyuhjtman — APPROVED** [review]

**Comment by johngrantuk** [comment]
Closed in favour of: https://github.com/balancer-labs/balancer-sor/pull/315

**Pull request closed without merging**